packages/@aws-cdk/aws-apigateway/lib/resource.ts

```diff
@@ -354,13 +365,13 @@ export abstract class ResourceBase extends ResourceConstruct implements IResourc
   }
 
   public get url(): string {
-    return this.restApi.urlForPath(this.path);
+    return this.api.urlForPath(this.path);
   }
 }
 
 export class Resource extends ResourceBase {
   public readonly parentResource?: IResource;
-  public readonly restApi: RestApi;
+  public readonly api: MutableRestApi;
```

Comment:
Is it possible to just expose `IRestApi` here and this way avoid exposing `MutableRestApi` as a public api?

This might require type checks in the code but it will keep the public api cleaner and with less surface.

Reply:
I've taken another stab at trying to erase this type from the public API. However, it requires some methods to be promoted into `IRestApi`. Let me know what you think here.